tests/unit/lastuser_core/test_model_client_AuthToken.py

```diff
@@ -0,0 +1,203 @@
+        """
+        Test for verifying creation of AuthToken instance
+        note: Only one authtoken per user and client
+        """
+        client = self.fixtures.client
+        crusoe = self.fixtures.crusoe
+        result = models.AuthToken(client=client, user=crusoe, scope=u'id', validity=0)
+        self.assertIsInstance(result, models.AuthToken)
+        self.assertEqual(result.user, crusoe)
+        self.assertEqual(result.client, client)
+
+    def test_authtoken_refresh(self):
+        """Test to verify creation of new token while retaining the refresh token."""
+        hagrid = models.User(username=u'hagrid', fullname=u'Rubeus Hagrid')
+        auth_token = models.AuthToken(user=hagrid, algorithm='hmac-sha-1')
+        existing_token = auth_token.token
+        existing_secret = auth_token.secret
+        auth_token.refresh()
+        self.assertNotEqual(existing_token, auth_token.token)
+        self.assertNotEqual(existing_secret, auth_token.secret)
+
+    def test_authtoken_is_valid(self):
+        """
+        Test for verifying if AuthToken's token is valid
+        """
+        client = self.fixtures.client
+        # scenario 1: when validity is unlimited (0)
+        tomriddle = models.User(username=u'voldemort', fullname=u'Tom Riddle')
+        scope=[u'id', u'email']
+        tomriddle_token = models.AuthToken(client=client, user=tomriddle, scope=scope, validity=0)
+        self.assertTrue(tomriddle_token.is_valid())
+
+        # scenario 2: when validity has not been given
+        draco = models.User(username=u'draco', fullname=u'Draco Malfoy')
+        draco_token = models.AuthToken(client=client, user=draco, scope=scope)
+        with self.assertRaises(TypeError):
+            draco_token.is_valid()
+
+        # scenario 3: when validity is limited
+        harry = models.User(username=u'harry', fullname=u'Harry Potter')
+        harry_token = models.AuthToken(client=client, user=harry, scope=scope, validity=3600, created_at=datetime.utcnow())
+        self.assertTrue(harry_token.is_valid())
+
+        # scenario 4: when validity is limited *and* the token has expired
+        cedric = models.User(username=u'cedric', fullname=u'Cedric Diggory')
+        cedric_token = models.AuthToken(client=client, user=cedric, scope=scope, validity=1, created_at=datetime.utcnow() - timedelta(1))
+        self.assertFalse(cedric_token.is_valid())
+
+    def test_AuthToken_get(self):
+        """
+        Test for retreiving a AuthToken instance given a token
+        """
+        specialdachs = self.fixtures.specialdachs
+        oakley = self.fixtures.oakley
+        scope = [u'id']
+        dachsadv = models.Client(title=u"Dachshund Adventures", org=specialdachs, confidential=True, website=u"http://dachsadv.com")
+        auth_token = models.AuthToken(client=dachsadv, user=oakley, scope=scope)
+        token = auth_token.token
+        db.session.add(dachsadv, auth_token)
+        result = models.AuthToken.get(token)
```

Comment:
If `get` is being tested here, `self.assertEqual(result, auth_token)` should be sufficient?